src/Interpreters/Cluster.cpp

```diff
@@ -813,6 +813,17 @@ Cluster::Cluster(Cluster::ReplicasAsShardsTag, const Cluster & from, const Setti
     secret = from.secret;
     name = from.name;
 
+    if (max_hosts > 0 && shards_info.size() > max_hosts)
```

Comment:
Shouldn't you also do something about:

`addresses_with_failover` and
`slot_to_shard`?

Reply:
Good point, I missed it.